config/initializers/omniauth.rb

```diff
@@ -16,7 +27,8 @@
     # * info[office_codes] => ["1A123B", "2A555X"]
     if FeatureFlags.omniauth_test_mode.enabled?
       config.before_callback_phase do |env|
-        env['omniauth.auth'].merge!(
+        AuthMemory.put(env['omniauth.auth'].dup)
+        env['omniauth.auth'] = AuthMemory.get.merge(
```

Comment:
I'm. a bit confused by this. What's the benefit of this compared to simply doing:

```
env['omniauth.auth'] = env['omniauth.auth'].dup.merge(
```
?

Reply:
So this is a fun fact about how the test mode appears to work as it caches what is set in `env['omniauth.auth']` between requests (at least in the tests)... hence I am storing the initial value in memory to avoid this glitch...

Reply:
Weird. For me the tests pass with the change I suggested above, as all it needed was the `dup` - I still don't see how assigning the `dup`'d value to an attribute and taking it out again would change things. But if you're seeing a problem without that extra step then fair enough.

Reply:
I'm still intrigued by this. When I [remove AuthMemory](https://github.com/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/commit/10b5adf948122f233f49b6c9223e9340c1b97b35), both locally and [on CI](https://app.circleci.com/pipelines/github/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/1221/workflows/761282a1-5cdd-47ec-b47f-d0d2f036f64d/jobs/6777) everything passes. Are you sure it's necessary on your local machine? Are you working in Docker?

Reply:
arrrg... assumptions... so `merge!` broke the tests so assumed setting the value would as well... but it doesn't... you don`t even need to dup....